src/liballoc/raw_vec.rs

```diff
@@ -736,6 +736,10 @@ fn alloc_guard(alloc_size: usize) -> Result<(), CollectionAllocErr> {
     }
 }
 
+fn capacity_overflow() -> ! {
```

Comment:
This feels non-obvious as to why it's extracted, maybe a comment would be good?

Reply:
👍